File: rust/azure_iot_operations_protocol/tests/metl/default_sender.rs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use serde::Deserialize;

use crate::metl::default_serializer::DefaultSerializer;

#[derive(Deserialize, Debug)]
pub struct DefaultSender {
    #[serde(rename = "telemetry-name")]
    pub telemetry_name: Option<String>,

    #[serde(rename = "serializer")]
    pub serializer: Option<DefaultSerializer>,

    #[serde(rename = "telemetry-topic")]
    pub telemetry_topic: Option<String>,

    #[serde(rename = "topic-namespace")]
    pub topic_namespace: Option<String>,
}

File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/DefaultSender.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class DefaultSender
    {
        public DefaultSender()
        {
            Serializer = new();
        }

        public string? TelemetryName { get; set; }

        public DefaultSerializer Serializer { get; set; }

        public string? TelemetryTopic { get; set; }

        public string? TopicNamespace { get; set; }
    }
}
